plumbing/format/packfile/decoder.go

```diff
@@ -105,6 +108,8 @@ func NewDecoderForType(s *Scanner, o storer.EncodedObjectStorer,
 
 		offsetToType: make(map[int64]plumbing.ObjectType, 0),
 		decoderType:  t,
+
+		cache: cache.NewObjectFIFO(cache.MaxSize),
```

Comment:
The `cache.MaxSize` should be added as external repository configuration. Some idea of how to do it correctly?